### DISCUSSION TITLE
How to to integrate Bevy Engine?

### DISCUSSION THREAD

**Discussion by greenmughal** [fintelia/terra#20] [General]
How to to integrate Bevy Engine? both projects support webgpu.

**Comment by fintelia** [comment]
I'm not really familiar with Bevy. However, it should be enough to just grab references to the wgpu::Device and a wgpu::Queue and pass them to Terra. It looks those two objects are stored inside the [WgpuRenderer struct](https://docs.rs/bevy/0.5.0/bevy/wgpu/struct.WgpuRenderer.html).

**Reply by greenmughal** [reply]
Thanks, well im new to both projects, still have to take a deep look in.

**Reply by fintelia** [reply]
Started poking around with this myself. First thing I noticed is that you need to use the git version of bevy because the last crates.io version depends on an old version of wgpu.

**Comment by greenmughal** [comment]
https://github.com/bevyengine/bevy/discussions/2210

Please see above topic.
My intent is to create atleast a proof of concept.

**Comment by fintelia** [comment]
Update: I did some more experimenting with this. Using https://github.com/bevyengine/bevy/pull/2240 and https://github.com/fintelia/terra/tree/bevy it is possible to patch something together. However, it seems that bevy is planning to transition to a new renderer that would break any more work here.

I plan to keep an eye on bevy development and see whether it is feasible to release a _terra_ plugin for it in the future.

**Comment by greenmughal** [comment]
Hi, any update on this one, bevy 0.6 looks few months away, any hopes for a plugin for bevy?

**Reply by fintelia** [reply]
My current plan is to wait for the bevy 0.6 release, and then assess feasibility again

**Comment by samcarey** [comment]
Hello, any update, now that 0.6 (and 0.7 in a few days) is out?

**Reply by fintelia** [reply]
Last time I checked this was blocked on https://github.com/bevyengine/bevy/pull/3595
